### PR TITLE
fix: ensure log formatting (JSON) is respected

### PR DESCRIPTION
**Pull request by davidby-influx**
Replace usage of the Go log package with the zap
logger and respect the formatting specified in
the configuration file.

closes https://github.com/influxdata/influxdb/issues/22104

- [X] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [X] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [X] Rebased/mergeable
- [X] Tests pass

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@
 -	[#21992](https://github.com/influxdata/influxdb/pull/21992): fix: systemd-start script should be executable by group and others
 -	[#22019](https://github.com/influxdata/influxdb/pull/22019): fix: error instead of panic when enterprise tries to restore with OSS
 -	[#22040](https://github.com/influxdata/influxdb/pull/22040): fix: copy names from mmapped memory before closing iterator
+-	[#22104](https://github.com/influxdata/influxdb/issues/22104): fix: ensure log formatting (JSON) is respected
```
Comment (wbaker85) [inline CHANGELOG.md:31]: I think this needs updated to link to the pull request instead of the issue.

Comment (davidby-influx) [reply CHANGELOG.md:31]: Whoops!

Comment (davidby-influx) [reply CHANGELOG.md]: Friday afternoon and too many browser tabs open...  Fixed.

**Review by wbaker85 — APPROVED** [review]
LGTM

**Pull request merged**